### ISSUE TITLE
why loss of landmark task does not descend?

### ISSUE THREAD

**Issue by gobigrassland** [foreverYoungGitHub/MTCNN#23]
I have trained your codes many times, but the loss of landmark task does not converge. I don' t known what is wrong.  When i only train face classification and regression of bounding boxes, losses of these tasks both descend. Why?

**Comment by foreverYoungGitHub** [comment]
Do you generate the training data and labels by yourself? 

If so, what's the learning rate of your code?

**Comment by gobigrassland** [comment]
I check and modify the code generating training data. 
Then i re-generate the training data and labels and  train again. Although the loss of landmark seems to converge, the detected landmarks of faces are not correct.

**Comment by KangolHsu** [comment]
i prepare my data and label like this:

image_path  face   face_box_cords          landmarks 
img1.jpg       1       0.12 0.21 0.34 0.27    [0.1,0.2,0.3,0.4,0.5,0.1,0.2,0.3,0.4,0.5];    //positive face
img2.jpg       1       0.02 0.12 0.24 0.17    [0.1,0.2,0.3,0.4,0.5,0.1,0.2,0.3,0.4,0.5];   // partial face
img3.jpg       0       0     0      0         0      [ 0,   0 ,   0,   0 ,  0 , 0 ,  0 ,  0 ,  0 , 0 ];  // no face
......
......

i want to konw  whether the image data (eg:img1.jpg ) is just a face image or  face part with background part?

**Comment by blankWorld** [comment]
@gobigrassland Hi,I met the same question . Have you solved the problem ? Although my landmarks loss converge at last it descend so slowly and it does badly on my landmark validation set.
